### PR TITLE
Fix segfault in case of no write access to log dir

### PR DESCRIPTION
**Pull request by jmachowinski**
# 🦟 Bug fix

Fixes a Segfault when building Ign common3 with pbuilder

## Summary
If there is no write access to the log directory, the logger will call itself recusively in an infinite loop and segfault.
Note, this bug is also present in the latest version of ign common.

### CODE REVIEW

**Comment by mjcarroll** [comment]
> Note, this bug is also present in the latest version of ign common.

once merged, this will be forward ported across versions.

File: src/Console.cc [hunk 1]
```diff
@@ -261,8 +261,8 @@ void FileLogger::Init(const std::string &_directory,
   {
     if (!env(IGN_HOMEDIR, logPath))
     {
-      ignerr << "Missing HOME environment variable."
-        << "No log file will be generated.";
+      std::cerr << "Missing HOME environment variable."
```
Comment (iche033) [inline src/Console.cc:264]: space between the two messages or new line?

```suggestion
      std::cerr << "Missing HOME environment variable. "
```

File: src/Console.cc [hunk 2]
```diff
@@ -275,7 +275,12 @@ void FileLogger::Init(const std::string &_directory,
   auto* buf = dynamic_cast<FileLogger::Buffer*>(this->rdbuf());
 
   // Create the directory if it doesn't exist.
-  createDirectories(logPath);
+  if(!createDirectories(logPath, std::cerr))
+  {
+      std::cerr << "Failed to generate log directories."
```
Comment (iche033) [inline src/Console.cc:280]:
```suggestion
      std::cerr << "Failed to generate log directories. "
```

**Comment by jmachowinski** [comment]
ping ?

File: src/EnumIface_TEST.cc [hunk 3]
```diff
@@ -41,7 +41,7 @@ IGN_ENUM(myTypeIface, MyType, MY_TYPE_BEGIN, MY_TYPE_END,
 /////////////////////////////////////////////////
 TEST_F(EnumIfaceTest, StringCoversion)
 {
-  MyType type;
+  MyType type = MyType::TYPE2;
```
Comment (mjcarroll) [inline src/EnumIface_TEST.cc:44]: Is this change related to the rest of the pull request?

Comment (jmachowinski) [reply src/EnumIface_TEST.cc:44]: No, this is a separate commit fixing a compile warning.
